python_modules/libraries/dagster-databricks/dagster_databricks_tests/test_pipes.py

```diff
@@ -62,8 +68,31 @@ def client() -> WorkspaceClient:
 
 TASK_KEY = "DAGSTER_PIPES_TASK"
 
+DAGSTER_PIPES_WHL_FILENAME = "dagster_pipes-1!0+dev-py3-none-any.whl"
+
 # This has been manually uploaded to a test DBFS workspace.
-DAGSTER_PIPES_WHL_PATH = "dbfs:/FileStore/jars/dagster_pipes-1!0+dev-py3-none-any.whl"
+DAGSTER_PIPES_WHL_PATH = f"dbfs:/FileStore/jars/{DAGSTER_PIPES_WHL_FILENAME}"
+
+
+def get_repo_root() -> str:
+    path = os.path.dirname(__file__)
+    while not os.path.exists(os.path.join(path, ".git")):
+        path = os.path.dirname(path)
+    return path
+
+
+@contextmanager
+def upload_dagster_pipes_whl(client: WorkspaceClient) -> Iterator[None]:
+    repo_root = get_repo_root()
+    orig_wd = os.getcwd()
+    dagster_pipes_root = os.path.join(repo_root, "python_modules", "dagster-pipes")
+    os.chdir(dagster_pipes_root)
+    subprocess.check_call(["python", "setup.py", "bdist_wheel"])
+    subprocess.check_call(
+        ["dbfs", "cp", "--overwrite", f"dist/{DAGSTER_PIPES_WHL_FILENAME}", DAGSTER_PIPES_WHL_PATH]
+    )
+    os.chdir(orig_wd)
+    yield
```

Comment:
what is going on here? can you comment?

Reply:
This is a fixture that uploads `dagster-pipes` as a WHL alongside the test script every time you run the target test. Added an explanatory comment.